pkg/infradb/port.go

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (c) 2022-2023 Dell Inc, or its subsidiaries.
+// Copyright (c) 2022-2023 Intel Corporation, or its subsidiaries.
+// Copyright (C) 2023 Nordix Foundation.
+
+package infradb
+
+import (
+	//	"fmt"
+	"log"
+	"net"
+
+	pb "github.com/opiproject/opi-api/network/evpn-gw/v1alpha1/gen/go"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb/common"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb/subscriberframework/eventbus"
+)
+
+// BridgePortType reflects the different types of a Bridge Port
+type BridgePortType int32
+
+const (
+	// Unspecified bridge port type
+	Unspecified BridgePortType = iota
+	// Access bridge port type
+	Access = iota
+	// Trunk bridge port type
+	Trunk = iota
+)
```

Comment:
We should be able to use the definitions for the Bridge Port Type from the VLAN protobuf that is in the review in the network area in the future to keep things common.  Also that various status indicators.

Reply:
In general we use the pb types. What you see here is the internal translation that happens from a PB object to an internal infradb domain object. So the protobufs types internally are translated to an infradb domain object which is more ruch and we can capture on that internal information that the user doesn't want to see but it is crucial fro the system to work. So we have chosen to do an internal translation to the BP type from the PB types to the internal infradb object. This way we will not need to impor the Protobufs everywhere but ony in the place where the actual transation happens. This transation from pb types ton the internal domain types (infradb) and vise versa is an architecture/design choise

Reply:
We may just want to track this for refactoring/optimization with the PB schemes for the future as an item.

Reply:
Sorry I didn't get that fully. Can you please explain what do you mean ?

Reply:
For the new PB types that will cover that type of operation (common protobufs), we can open an issue to track the future change that would handle this from a common perspective as well as any definitions in order to simplify the definitions and not have them spread and duplicated in various locations.  Nothing to do for now until the common protobufs are reviewed/merged into the repo.  Once they are merged, then we open an issue to track migration to a common definition.